data_source_alks_session.go

```diff
@@ -33,13 +36,19 @@ func dataSourceAlksSessionRead(ctx context.Context, d *schema.ResourceData, meta
 	client := providerStruct.client
 	credentials := client.Credentials.(interface{})
 
-	stsCredentials, ok := credentials.(alks.STS)
+	// Type assertion: Since STS is a struct, we need to cast it to the interface.
+	stsCredentials, ok := credentials.(*alks.STS)
 
 	if ok {
 		_ = d.Set("access_key", stsCredentials.AccessKey)
 		_ = d.Set("secret_key", stsCredentials.SecretKey)
 		_ = d.Set("session_token", stsCredentials.SessionToken)
+		// When the below line was not present I could not get credentials to be returned
+		// as terraform outputs during runtime.
+		d.SetId(client.AccountDetails.Account)
 	}
 
+	// If not ok, we return empty values for credentials and move along.  Is this what we want?
```

Comment:
I would think best to fail fast, if possible. Whatever you’re using these credentials for downstream is not going to work if this outputs blank values, so raising an exception here might help triangulate the problem faster. I just did not know how to raise errors in Go/Terraform providers.